### ISSUE TITLE
With-apollo does not fetch on the server when using url.query

### ISSUE THREAD

**Issue by nickluger** [zeit/next.js#3603]
- [x] I have searched the [issues](https://github.com/zeit/next.js/issues) of this repository and believe that this is not a duplicate.

## In short

Using with-apollo with `url` or `query` params in pages leads to strange behaviour.

## What works:

I extended the `index.js` in the _with-apollo_-example, just to print the current query:

```js
export default withData(props => (
  <Main>
    {(props.url) && console.log(props.url.query)} // THIS WILL BECOME IMPORTANT!
    <Header />
    <Submit />
    <PostList />
  </Main>
))
```
The apollo data (the posts) are fetched correctly **on the server** and on the client i get them baked into `__NEXT_DATA__ as `apollo store`.

```js
  __NEXT_DATA__ = {"props":{"serverState":{"apollo":{"data":{"Post:cjcq65k4ua4h801287p9h0rw9":{"id":"cjcq65k4ua4h801287p9h0rw9","titl
```
## What does not work:
 
```js
export default withData(props => (
  <Main>
    {console.log(props.url.query)} // JUST A MINOR CHANGE..?
    <Header />
    <Submit />
    <PostList />
  </Main>
))
```
Now on the client the apollo store is **empty** and data gets fetched **on the client**:
```js
__NEXT_DATA__ = {"props":{"serverState":{"apollo":{"data":{}}}},
```
## Steps to Reproduce (for bugs)
1. Get an Apollo next repo like https://github.com/ads1018/next-apollo-example
2. `console.log()` the query like above
3. See loading state on client

## What am i trying to accomplish?

I want to **export a static** app with routes like:

```
/posts/My-first-post
/posts/My-other-post
```
where posts are known at compile time. For that i fetch all post ids at export-time/compile-time in `next.config.js`  and map them to pages with variables, in the `exportPathMap`-function, like this:
```js
// ...
routes[`/posts/${slug(post.title)}`] = {page: '/posts', query: {post_id: post.id}}
// ...
return routes
```
I use graphql wrappers and use the query parameter to fetch the actual data, and the `next-apollo` package to get the data in `getInitialProps`:
```js
graphql(
 PostDataQuery,
  {
    options: ({url}) => {
        return {variables: {id: url.query.post_id}}
  }
)(MyPostComponent)
```
But as described above, the data i loaded on the client and not on the server if **i use query parameters**, and the apollo store is not created or not sent to the client.

For example this works:

```js
// ... 
        return {variables: {id: 'some-hard-coded-id'}}
// ...
```

## Your Environment

| Tech    | Version |
|---------|---------|
| next    |      v4.2.3    |
| node    |   v9.2.0      |
| OS      |    MacOS     |
| browser |    Chrome     |
| etc     |         |

**Comment by tomsvogel** [comment]
same problem here

**Comment by nickluger** [comment]
I compared two version of the `withApollo`-file.

The first is defined in the `next-apollo`-library you can find here: https://github.com/ads1018/next-apollo, by @ads1018 . 

It contains the following code:
```js
 if (!process.browser) {
        const apollo = initApollo()

        try {
          // Run all GraphQL queries
```

The other version is used in the `with-apollo` example, here in Next's repo.  (The example uses almost the same code, of course it's also by @ads1018 (Thank you for integrating Apollo, by the way 👍👍👍) :
```js
if (!process.browser) {
        const apollo = initApollo()
        // Provide the `url` prop data in case a GraphQL query uses it 
        const url = {query: ctx.query, pathname: ctx.pathname} // THAT's IT!
```

You need to use the second version, to make url work.

**Comment by ads1018** [comment]
Thanks @nickluger. Looks like a breaking change was introduced into the `next-apollo` library last month. https://github.com/ads1018/next-apollo/pull/7/files.

I'll patch that up. Apologies for the confusion!

**Comment by hobbs** [comment]
@ads1018 did that patch ever make it into next-apollo?

**Comment by ads1018** [comment]
Apologies for the delay on this. Just published an update (1.0.11) so you have access to the url prop again.

**Comment by ads1018** [comment]
![highfive](https://user-images.githubusercontent.com/555740/36009493-4f6c4e4e-0d1a-11e8-80d0-9d9cbe2a9560.gif)
